eessi/testsuite/hooks.py

```diff
@@ -282,16 +282,54 @@ def _assign_one_task_per_gpu(test: rfm.RegressionTest):
     log(f'num_cpus_per_task set to {test.num_cpus_per_task}')
     log(f'num_tasks set to {test.num_tasks}')
 
+def _set_or_append_valid_systems(test: rfm.RegressionTest, valid_systems: str):
+    """
+    Sets test.valid_systems based on the valid_systems argument.
+    - When valid_systems is an empty string, test.valid_systems will be set equal to eessi.testsuite.constants.INVALID_SYSTEM
+    - When no test.valid_system was set yet, or it was at the default value ['*'], it will be overwritten by [valid_system]
+    - When a test.valid_system was already set, valid_system will be appended to it. This allows adding requests for multiple partition features by different hooks.
```

Comment:
lines are a bit too long
```suggestion
    - If valid_systems is an empty string, test.valid_systems is set equal to eessi.testsuite.constants.INVALID_SYSTEM
    - If no test.valid_system was set yet, or it was at the default value ['*'], it is overwritten by [valid_system]
    - If a test.valid_system was already set, valid_system is appended to it, which allows adding requests for multiple
    partition features by different hooks.
```